File: benchmark/ruby/strrev.rb
start = Time.now.to_f * 1000
s = "a" * 50000
rev = ""
(s.length - 1).downto(0) do |j|
  rev += s[j]
end
finish = Time.now.to_f * 1000
puts "Len: #{rev.length}"
puts "Time: #{(finish - start).round}ms"

File: benchmark/php/strrev.php
<?php
$s = str_repeat("a", 50000);
$start = hrtime(true);
$rev = "";
for ($j = strlen($s) - 1; $j >= 0; $j--) { $rev .= $s[$j]; }
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Len: " . strlen($rev) . "
";
echo "Time: " . round($ms) . "ms
";
